### PR TITLE
no synchronous function call force anymore and some test fixes

### PR DESCRIPTION
**Pull request by Syzuna**
This PR is an attempt to reduce the possibility of deadlocks in UI applications like WinForms or WPF.
We had some internal code that for no obvious reason forced async HTTP calls to run synchronously and causing issues with said UI applications.

With that PR I also fixed the Tests to see if my changes broke anything but all Tests came back green.

https://cdn.syzuna-programs.de/images/Mjxe3M6zHG.png

### CODE REVIEW

File: TwitchLib.Api.Core/HttpCallHandlers/TwitchHttpClient.cs [hunk 1]
```diff
@@ -98,7 +99,7 @@ public int RequestReturnResponseCode(string url, string method, List<KeyValuePai
                 RequestUri = new Uri(url),
                 Method = new HttpMethod(method)
             };
-            var response = _http.SendAsync(request).GetAwaiter().GetResult();
+            var response = await _http.SendAsync(request);
```
Comment (dave-sekula) [inline TwitchLib.Api.Core/HttpCallHandlers/TwitchHttpClient.cs:102]: just need to add `.ConfigureAwait(false)` to each of the async calls to return to the correct context :)

**Review by dave-sekula — CHANGES_REQUESTED** [review]
.ConfigureAwait(false) on any of the async awaits please
(I'm on my other account on phone :D / Prom.)

**Comment by Syzuna** [comment]
did I miss some? will add them

File: TwitchLib.Api.Core/HttpCallHandlers/TwitchHttpClient.cs (hunk 1 above)
Comment (Syzuna) [reply TwitchLib.Api.Core/HttpCallHandlers/TwitchHttpClient.cs]: Hope I catched all now

**Comment by Syzuna** [comment]
I should have catched all missing ones I hope @dave-sekula

File: TwitchLib.Api.Core/Undocumented/Undocumented.cs [hunk 2]
```diff
@@ -170,16 +170,16 @@ public Task<ChatUserResponse> GetChatUserAsync(string userId, string channelId =
 
         #region IsUsernameAvailable
 
-        public Task<bool> IsUsernameAvailableAsync(string username)
+        public async Task<bool> IsUsernameAvailableAsync(string username)
         {
             var getParams = new List<KeyValuePair<string, string>> {new KeyValuePair<string, string>("users_service", "true")};
-            var resp = RequestReturnResponseCode($"https://passport.twitch.tv/usernames/{username}", "HEAD", getParams);
+            var resp = await RequestReturnResponseCode($"https://passport.twitch.tv/usernames/{username}", "HEAD", getParams);
```
Comment (prom3theu5) [inline TwitchLib.Api.Core/Undocumented/Undocumented.cs:176]: just one more ConfigureAwait(false) here

**Review by prom3theu5 — CHANGES_REQUESTED** [review]
LGTM after this one :)
Nice one man

File: TwitchLib.Api.Core/Undocumented/Undocumented.cs (hunk 2 above)
Comment (Syzuna) [reply TwitchLib.Api.Core/Undocumented/Undocumented.cs:176]: x.x

I rly need to setup analyzers for that xD

**Review by prom3theu5 — APPROVED** [review]

**Pull request merged**